src/datasets/tasks/automatic_speech_recognition.py

```diff
@@ -1,20 +1,24 @@
 from dataclasses import dataclass
 from typing import ClassVar, Dict
 
-from ..features import Features, Value
+from ..features import Audio, Features, Value
 from .base import TaskTemplate
 
 
 @dataclass(frozen=True)
 class AutomaticSpeechRecognition(TaskTemplate):
     task: str = "automatic-speech-recognition"
-    # TODO(lewtun): Replace input path feature with dedicated `Audio` features
-    # when https://github.com/huggingface/datasets/pull/2324 is implemented
-    input_schema: ClassVar[Features] = Features({"audio_file_path": Value("string")})
+    input_schema: ClassVar[Features] = Features({"audio": Audio(_storage_dtype="string")})
     label_schema: ClassVar[Features] = Features({"transcription": Value("string")})
-    audio_file_path_column: str = "audio_file_path"
+    audio_column: str = "audio"
     transcription_column: str = "transcription"
+    # set the `streaming` flag when casting streamable datasets that have `Audio` stored as `struct{path, bytes}`
+    streaming: bool = False
+
+    def __post_init__(self):
+        if self.streaming:
+            self.__dict__["input_schema"] = Features({"audio": Audio(_storage_dtype="struct")})
```

Comment:
This is needed to support both the extracted and streamable datasets, avoiding type casting errors between `string` and `struct{}`. 
Maybe there's a better way to infer the underlying `Audio` storage type and not try to cast `Audio(_storage_dtype="struct")` <-> `Audio(_storage_dtype="string")`? Because setting the appropriate `streaming` flag could be confusing for contributors without additional docs.

To reproduce the type casting error, just remove this `__post_init__()` method and try converting CommonVoice:
```python
load_dataset("datasets/common_voice", "tr", split="train").prepare_for_task("automatic-speech-recognition")
```

Reply:
If I'm not mistaken, we can avoid this logic by updating the `prepare_for_task` method to not call `cast` on the complex feature types (Audio, Image) with the `decode_example` method defined and to call `cast_column` on them instead.

Ideally, `cast` should know how to handle complex types (Audio, Image, ArrayXD), but currently, it doesn't.